### PR TITLE
Fix for invalid JSON string (due to duplicate keys) when using `@JsonAnyGetter`

### PR DESCRIPTION
**Pull request by Bishnup1995**

### CODE REVIEW

**Comment by Bishnup1995** [comment]
The pull request is related to [https://github.com/FasterXML/jackson-databind/issues/4091](this) issue.

**Comment by Bishnup1995** [comment]
@JooHyukKim Please review the changes after the latest commit. Thank you for your response

**Comment by Bishnup1995** [comment]
@JooHyukKim I have already tried using `STRICT_DUPLICATE_DETECTION` but no luck. Please review the changes.

File: src/main/java/com/fasterxml/jackson/databind/ser/std/MapSerializer.java [hunk 1]
```diff
@@ -913,6 +933,41 @@ public void serializeFieldsUsing(Map<?,?> value, JsonGenerator gen, SerializerPr
         }
     }
 
+    public void serializeFieldsUsing(Map<?, ?> value, JsonGenerator gen, SerializerProvider provider,
+                                     JsonSerializer<Object> ser, Object bean)
+            throws IOException {
+        final JsonSerializer<Object> keySerializer = _keySerializer;
+        final TypeSerializer typeSer = _valueTypeSerializer;
+        Set<String> fields = Arrays.stream(bean.getClass().getDeclaredFields()).map(Field::getName).collect(Collectors.toSet());
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/ser/std/MapSerializer.java:941]: I am sorry but this is not going to work for most cases plus adds significant overhead for every usage.

**Comment by cowtowncoder** [comment]
While I appreciate PRs, this is not something I would consider accepting: it is not possible to simply try to do a lookup of fields or methods of a class to try to determine which JSON properties would be written.
To properly determine POJO properties, full introspection is needed (possibly getting hold of `BeanSerializer`). That would not address problem of custom serializers; and then there's the question of filters (for `@JsonView`, `@JsonFilter`).

**Comment by cowtowncoder** [comment]
Not suitable to be used in its current form unfortunately, closing.

**Pull request closed without merging**